### ISSUE TITLE
[Spec]  My experience with trying to use CollectionView and SwipeView - and a request

### ISSUE THREAD

**Issue by Tommigun1980** [xamarin/Xamarin.Forms#12599]
Hi. I really like and appreciate Xamarin.Forms but I wanted to detail my experience with trying to use a new component.

My use case is simple - I want to display chat messages in some kind of list form, where individual chat messages can be deleted. So far I've been using a bindable StackLayout but wanted to replace it with a proper virtualising list as I'm nearing release.

CollectionView is no longer marked as experimental, but this is the amount of bugs I encountered for this simple use case:
https://github.com/xamarin/Xamarin.Forms/issues/12555 (a total blocker)
https://github.com/xamarin/Xamarin.Forms/issues/12554 (a total blocker)
https://github.com/xamarin/Xamarin.Forms/issues/12576 (this I found a workaround for)
https://github.com/xamarin/Xamarin.Forms/issues/12553 (this just breaks hot reload so is not a blocker)

SwipeView is still experimental but these are blockers for me:
https://github.com/xamarin/Xamarin.Forms/issues/12518 (four issues in this report that are closely related and totally blockers)

I thought I'd use a ListView instead (which has been deprecated but largely works), but ran into:
https://github.com/xamarin/Xamarin.Forms/issues/12598 (may be caused by hot reload so not necessarily a blocker)

The reason I even thought about using ListView is that the memory leaks seem to finally have been fixed with PR https://github.com/xamarin/Xamarin.Forms/pull/12535

So all in all -- I am in a total pinch here until at least https://github.com/xamarin/Xamarin.Forms/issues/12555, https://github.com/xamarin/Xamarin.Forms/issues/12554 and https://github.com/xamarin/Xamarin.Forms/issues/12518 are fixed.

I find it quite disheartening having to spend this much time testing every single component just to find several blockers in them as well, in my very-much-not-an-advanced-use-case. Something that should have taken a very limited amount of time turned into days of investigation and me filling documents with alternative paths to take depending on what gets fixed.

I think most CollectionView issues have the root cause https://github.com/xamarin/Xamarin.Forms/issues/11511 which is caused by https://github.com/xamarin/Xamarin.Forms/issues/10842, so fixing that one thing only may make it a usable component. Just fixing two of the issues in (https://github.com/xamarin/Xamarin.Forms/issues/12518), that SwipeView constantly opens accidentally as it's too sensitive, and then proceeds to lock the UI completely, would make it usable.

But as it stands right now I am completely at the mercy of these getting fixed, and the MeasureAllItems issue in CollectionView has been open for half a year. So I am begging you to fix it to make CollectionView usable, or provide some guidance on how to get to release (this post details what I've tried so far).

Thank you so much for creating this technology and supporting it!

**Comment by transis2** [comment]
I agree with you, threre's so much issues being filled than PR to fix them.
They desesperatly need to hire more devs to help the team
2.6K issues open and growing...

**Comment by daves1992** [comment]
I agree, Xamarin is a brilliant platform and I use it quite heavily for my companies mobile apps but some bugs have been open for years. It also seems just browsing the github projects it can take months from requesting reviews to actually getting them so they must need more devs.
